tools/runners/run-mongo.js

```diff
@@ -657,62 +657,29 @@ var launchMongo = function (options) {
         return;
       }
 
-      let wasJustSecondary = false;
+      let writableTimestamp = Date.now();
 
-      // XXX timeout eventually?
+      // Wait until the primary is writable. If it isn't writable after one
+      // minute, throw an error and report the replica set status.
       while (!stopped) {
-        var status = yieldingMethod(
-          db.admin(), 'command', {replSetGetStatus: 1});
-
-        // See https://docs.mongodb.com/manual/reference/replica-states/
-        // for information on various states
-
-        // Are any of the members starting up or recovering?
-        if (_.any(status.members, function (member) {
-          return member.stateStr === 'STARTUP' ||
-            member.stateStr === 'STARTUP2' ||
-            member.stateStr === 'RECOVERING';
-        })) {
-          utils.sleepMs(20);
-          continue;
-        }
-
-        const firstMemberState = status.members[0].stateStr;
-
-        // Is the intended primary currently a secondary? (It passes through
-        // that phase briefly.)
-        if (firstMemberState === 'SECONDARY') {
-          utils.sleepMs(20);
-          wasJustSecondary = true;
-          continue;
-        }
-
-        // Mongo 3.2 introduced a new heartbeatIntervalMillis property
-        // on replica sets, used during "primary" negotiation.
-        //
-        // If the first member was _just_ promoted, we'll wait until
-        // the heartbeat interval has elapsed before proceeding since
-        // the decision is not official until the heartbeat has elapsed.
-        if (firstMemberState === 'PRIMARY' && wasJustSecondary) {
-          wasJustSecondary = false;
-          utils.sleepMs(status.heartbeatIntervalMillis);
-          continue;
-        }
+        const { ismaster } = yieldingMethod(db.admin(), "command", {
+          isMaster: 1
```

Comment:
In case anyone is wondering, the `isMaster` command has been supported since well before Mongo 3.2.22, which is [the version we use for 32-bit systems](https://github.com/meteor/meteor/blob/30a19139431d48cb3caf913dd2fb7748768291dd/scripts/build-dev-bundle-common.sh#L10): https://docs.mongodb.com/v2.2/reference/command/isMaster/